File: algorithms/beautiful_days_at_the_movies.hs
main :: IO ()
main = interact $ show . solve . map read . words

solve :: [Int] -> Int
solve [i, j, k] = length $ filter isBD [i..j]
  -- where isBD d = (abs $ d - (read $ reverse $ show d)) `mod` k == 0
  where isBD d = (==0) . (`mod` k) . abs $ d - read (reverse $ show d)
solve _ = error "wrong input"

File: hackerrank/algorithms/beautiful_days_at_the_movies.clj
(require '[clojure.string :as s])

(defn isBD? [d k]
  (= 0 (mod (Math/abs (- d (Integer/parseInt (s/reverse (str d)))))
            k)))

(let [[i j k] (mapv read-string (s/split (read-line) #" "))]
  (prn (count (filter #(isBD? % k)
                      (range i (inc j))))))
